src/containers/rbtree/rbtree.h

```diff
@@ -0,0 +1,794 @@
+#include <algorithm>
+#include <iomanip>
+#include <iostream>
+template <typename K, typename T>
+class rbtree {
+ public:
+  using key_type = K;
+  using mapped_type = T;
+  using value_type = std::pair<const key_type, mapped_type>;
+  using reference = value_type&;
+  using const_reference = const value_type&;
+  using size_type = size_t;
+  using pointer = value_type*;
+  struct rbtree_node {
+    rbtree_node(const value_type& value)
+        : value_(value),
+          right_(nullptr),
+          left_(nullptr),
+          parent_(nullptr),
+          mColor('R') {}
```

Comment:
Рекомендую присмотреться к такому стилю форматирования списка инициализации в конструкторах.

Конечно, так или иначе всё отформатирует `clang-format`, но на время разработки можно своего стиля придерживаться, так как `clang-format` не всегда способствует удобочитаемости и комфорту работы.

```suggestion
    rbtree_node(const value_type& value)
        : value_(value)
        , right_(nullptr)
        , left_(nullptr)
        , parent_(nullptr)
        , mColor('R')
    {}
```